packages/themes/scss/_variables.scss

```diff
@@ -0,0 +1,285 @@
+  disabled-03: #8c8c8c,
+  highlight: #c9deff,
+  brand-01: #0062ff,
+  brand-02: #171717,
+  brand-03: #0062ff,
+  active-01: #bebebe,
+  hover-field: #e5e5e5,
+) !default !global;
+$theme--g10: $carbon--theme--g10;
+
+$carbon--theme--g90: (
+  interactive-01: #0062ff,
+  interactive-02: #6f6f6f,
+  interactive-03: #ffffff,
+  interactive-04: #408bfc,
+  ui-background: #282828,
+  ui-01: #3d3d3d,
+  ui-02: #565656,
+  ui-03: #565656,
+  ui-04: #8c8c8c,
+  ui-05: #f3f3f3,
+  text-01: #f3f3f3,
+  text-02: #bebebe,
+  text-03: #8c8c8c,
+  text-04: #ffffff,
+  icon-01: #f3f3f3,
+  icon-02: #bebebe,
+  icon-03: #ffffff,
+  link-01: #6ea6ff,
+  field-01: #3d3d3d,
+  field-02: #565656,
+  inverse-01: #171717,
+  inverse-02: #f3f3f3,
+  support-01: #fb4b53,
+  support-02: #3dbb61,
+  support-03: #fdd13a,
+  support-04: #408bfc,
+  overlay-01: rgba(23, 23, 23, 0.7),
+  focus: #ffffff,
+  hover-primary: #0353e9,
+  active-primary: #0530ad,
+  hover-primary-text: #054ada,
+  hover-secondary: #606060,
+  active-secondary: #3d3d3d,
+  hover-tertiary: #f3f3f3,
+  active-tertiary: #bebebe,
+  hover-ui: #353535,
+  active-ui: #6f6f6f,
+  selected-ui: #565656,
+  hover-selected-ui: #656565,
+  hover-danger: #ba1b23,
+  active-danger: #750e13,
+  hover-row: #4c4c4c,
+  visited-link: #bb8eff,
+  disabled-01: #3d3d3d,
+  disabled-02: #565656,
+  disabled-03: #8c8c8c,
+  highlight: #061f80,
+  brand-01: #0062ff,
+  brand-02: #6f6f6f,
+  brand-03: #ffffff,
+  active-01: #6f6f6f,
+  hover-field: #353535,
+) !default !global;
+$theme--g90: $carbon--theme--g90;
+
+$carbon--theme--g100: (
+  interactive-01: #0062ff,
+  interactive-02: #6f6f6f,
+  interactive-03: #ffffff,
+  interactive-04: #408bfc,
+  ui-background: #171717,
+  ui-01: #282828,
+  ui-02: #3d3d3d,
+  ui-03: #3d3d3d,
+  ui-04: #6f6f6f,
+  ui-05: #f3f3f3,
+  text-01: #f3f3f3,
+  text-02: #bebebe,
+  text-03: #6f6f6f,
+  text-04: #ffffff,
+  icon-01: #f3f3f3,
+  icon-02: #bebebe,
+  icon-03: #ffffff,
+  link-01: #6ea6ff,
+  field-01: #282828,
+  field-02: #3d3d3d,
+  inverse-01: #171717,
+  inverse-02: #f3f3f3,
+  support-01: #fb4b53,
+  support-02: #3dbb61,
+  support-03: #fdd13a,
+  support-04: #408bfc,
+  overlay-01: rgba(23, 23, 23, 0.7),
+  focus: #ffffff,
+  hover-primary: #0353e9,
+  active-primary: #0530ad,
+  hover-primary-text: #054ada,
+  hover-secondary: #606060,
+  active-secondary: #3d3d3d,
+  hover-tertiary: #f3f3f3,
+  active-tertiary: #bebebe,
+  hover-ui: #353535,
+  active-ui: #565656,
+  selected-ui: #3d3d3d,
+  hover-selected-ui: #4c4c4c,
+  hover-danger: #ba1b23,
+  active-danger: #750e13,
+  hover-row: #353535,
+  visited-link: #bb8eff,
+  disabled-01: #282828,
+  disabled-02: #3d3d3d,
+  disabled-03: #6f6f6f,
+  highlight: #061f80,
+  brand-01: #0062ff,
+  brand-02: #6f6f6f,
+  brand-03: #ffffff,
+  active-01: #565656,
+  hover-field: #353535,
+) !default !global;
+$theme--g100: $carbon--theme--g100;
+
+$interactive-01: map-get($carbon--theme, interactive-01) !default !global;
```

Comment:
The variables were previously set in a mixin. In `carbon-components`, the previous mixin was called inside of a feature flag mixin. With these variables not set from inside a mixin, how would one conditionally use these variables? With them not in a mixin, the `@import` is setting the variables from carbon-elements regardless if the `components-x` feature flag is set or not.

```scss
@import './vendor/@carbon/elements/scss/themes/themes';

@mixin theme--experimental() {
  // Set theme variables here
}
```

Reply:
Related to not defining variables from within a mixin, if you wanted to set an alternate default theme, using `$carbon--theme` you would:

```scss
@import '../../src/globals/scss/vendor/@carbon/elements/scss/themes/variables';

$carbon--theme: $carbon--theme--g90;

@import '../../src/globals/scss/styles';
```

First you'd have to import the theme variables, so then you could globally set `$carbon--theme`. This original PR had the global override be a string like `$carbon--theme: 'g90';` so no preliminary imports were needed. Think we need to refactor to prevent the preliminary import?

This snippet **only** works if the variables are defined in a mixin, and that mixin is called **once** in `_theme-tokens.scss` since the variables are `!default`ed.